interfaces/apparmor/spec.go

```diff
@@ -190,6 +191,34 @@ func (spec *Specification) setScope(securityTags []string) (restore func()) {
 	}
 }
 
+var metadataTagsSupported = apparmor_sandbox.MetadataTagsSupported
+
+// MetadataTagSnippet wraps the given AppArmor rule snippet in the given
+// metadata tags if tagging is supported, and returns the resulting snippet.
+// If tagging is not supported, returns the snippet unchanged, without error.
+// If any of the tags contain parentheses or AppArmor regular expression
+// characters, returns an error.
+//
+// For example, MetadataTagSnippet("/foo r,", []string{"bar", "baz"}) would
+// return `tags=(bar,baz) { /foo r, }`.
+func MetadataTagSnippet(snippet string, tags []string) (string, error) {
+	if len(tags) == 0 || !metadataTagsSupported() {
+		return snippet, nil
+	}
+	for _, tag := range tags {
+		if strings.ContainsAny(tag, "()") {
```

Comment:
I think we should put a way stricter limit and then gradually open it up if necessary. I kind of think you were going for the right thing with the dual check "not a regular expression" and "no parenthesis" but I think this should be a dedicated checker that literally says "this is a valid, single, apparmor tag name".